src/hotspot/cpu/ppc/templateTable_ppc_64.cpp

```diff
@@ -3803,15 +3803,18 @@ void TemplateTable::_new() {
     __ sldi(Roffset, Rindex, LogBytesPerWord);
     __ load_resolved_klass_at_offset(Rcpool, Roffset, RinstanceKlass);
 
-    // Make sure klass is fully initialized and get instance_size.
-    __ lbz(Rscratch, in_bytes(InstanceKlass::init_state_offset()), RinstanceKlass);
+    // Make sure klass is initialized.
+    assert(VM_Version::supports_fast_class_init_checks(), "Optimization requires support for fast class initialization checks");
+    __ clinit_barrier(Rcpool, R16_thread, nullptr /*L_fast_path*/, &Lslow_case);
```

Comment:
`Rcpool` is the wrong parameter. You need to use `RinstanceKlass`.

Reply:
Thank you for the correction @TheRealMDoerr . I have applied the code change. Tier1 test is running. I will push once it is done.

Reply:
build successful and no failures in tier1